packages/client/hmi-client/src/workflow/workflow.ts

```diff
@@ -0,0 +1,72 @@
+// Defines the type of data an operation can consume and output
+export interface OperationData {
+	type: string;
+	label?: string;
+}
+
+// Defines a function: eg: model, simulate, calibrate
+export interface Operation {
+	name: string;
+	description: string;
+	action: Function;
+	validation?: Function;
+
+	inputs: OperationData[];
+	outputs: OperationData[];
+}
+
+// Defines the data-exchange between WorkflowNode
+// In most cases the value here will be an assetId
+export interface WorkflowPort {
+	type: string;
+	value?: any;
+}
+
+// Node definition in the workflow
+// This is the graphical operation of the operation defined in operationType
+export interface WorkflowNode {
+	id: string;
+	workflowId: string;
+	operationType: string;
+
+	// Position on canvas
+	x: number;
+	y: number;
+	width: number;
+	height: number;
+
+	inputs: WorkflowPort[];
+	outputs: WorkflowPort[];
+
+	// FIXME: The section below is slated to be further spec'ed out later.
+	// State and progress, tracking of intermediate results
+	statusCode?: string; // not-run, completed, in-progress, error, invalid
+	intermediateIds?: WorkflowPort[];
+}
+
+export interface WorkflowEdge {
+	id: string;
+	workflowId: string;
+	points: { x: number; y: number }[];
+
+	source: string;
+	sourcePort: number;
+
+	target: string;
+	targetPort: number;
```

Comment:
```suggestion
	source: WorkflowNode["id"];
	sourcePort: number; // Are we really using integer like Petri Net?

	target: WorkflowNode["id"];
	targetPort: number;
```

Reply:
Nope. This is just direct numbering. Petrinet ascet's number system is indirect. e.g. to get value of index-2 in ascet is like `X[Y[2]]`

Reply:
This is going to be so annoying